marklogic-data-hub/src/testFixtures/java/com/marklogic/hub/test/AbstractHubTest.java

```diff
@@ -100,15 +104,41 @@ protected void resetDatabases() {
         runAsAdmin();
         String xquery = "cts:uris((), (), cts:not-query(cts:collection-query('hub-core-artifact'))) ! xdmp:document-delete(.)";
         HubClient hubClient = getHubClient();
-        hubClient.getStagingClient().newServerEval().xquery(xquery).evalAs(String.class);
-        hubClient.getFinalClient().newServerEval().xquery(xquery).evalAs(String.class);
-        hubClient.getJobsClient().newServerEval().xquery(xquery).evalAs(String.class);
-
-        HubConfig hubConfig = getHubConfig();
-        hubConfig.getAppConfig().newAppServicesDatabaseClient(hubConfig.getDbName(DatabaseKind.STAGING_SCHEMAS))
-            .newServerEval().xquery(xquery).evalAs(String.class);
-        hubConfig.getAppConfig().newAppServicesDatabaseClient(hubConfig.getDbName(DatabaseKind.FINAL_SCHEMAS))
-            .newServerEval().xquery(xquery).evalAs(String.class);
+        // Running these with primitive retry support, as if a connection cannot be made to ML, this is typically the
+        // first thing that will fail
+        retryIfNecessary(() -> hubClient.getStagingClient().newServerEval().xquery(xquery).evalAs(String.class));
+        retryIfNecessary(() -> hubClient.getFinalClient().newServerEval().xquery(xquery).evalAs(String.class));
+        retryIfNecessary(() -> hubClient.getJobsClient().newServerEval().xquery(xquery).evalAs(String.class));
+
+        DatabaseManager databaseManager = new DatabaseManager(hubClient.getManageClient());
+        databaseManager.clearDatabase(hubClient.getDbName(DatabaseKind.STAGING_SCHEMAS));
+        databaseManager.clearDatabase(hubClient.getDbName(DatabaseKind.FINAL_SCHEMAS));
+    }
+
+    /**
+     * Adding this to handle some intermittent connection failures that occur when Jenkins runs the tests.
+     * @param r
+     */
+    protected void retryIfNecessary(Runnable r) {
```

Comment:
Should we create our own no-arg/no-return functional interface to avoid confusion related to Runnable being related to threads?
https://stackoverflow.com/a/23958916/4460877

Reply:
I'm hoping I can remove this soon - it's just here for now to see if it helps at all. ML app servers shouldn't be unavailable for stretches of time during the test suite run - this is just to see if it helps. Hopefully the real problem can be fixed.

Reply:
I see. If we do need to add this in, I think it would be worthwhile to add a comment explaining that Runnable is only being used as a no-arg/no-return functional interface and does not have anything to do with Threads.